### PR TITLE
Rename the record parameter when its property get renamed

### PR DESCRIPTION
**Pull request by Cosifne**
Fix https://github.com/dotnet/roslyn/issues/72581

Reason:
Using code snnipet
```
    public record MyRecord(string MyProperty);
    public class ReferenceClass
    {
        public static void Test()
        {
            var record = new MyRecord("HelloWorld");
            var c = record.MyProperty;
        }
    }
```
When we rename the `MyProperty` in constructor, we are renaming the parameter, `var c = record.MyProperty` is not renamed, because it's referencing the generated property.

This PR fix by adding an additional check.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs [hunk 1]
```diff
@@ -121,6 +128,18 @@ static bool IsConstructorForType(ISymbol possibleConstructor, ISymbol possibleTy
...
+        private static bool IsPropertyGeneratedFromPrimaryConstructorParameter(
+            ISymbol propertySymbol, ISymbol parameterSymbol, CancellationToken cancellationToken)
+        {
+            if (parameterSymbol is IParameterSymbol parameter)
+            {
+                var generatedPropertySymbol = parameter.GetAssociatedSynthesizedRecordProperty(cancellationToken);
+                return propertySymbol.Equals(generatedPropertySymbol);
+            }
+
+            return false;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs:140]: this doesn't seem like the right fix to me.  Instead, shoudln't we be cascading between these members in the FIndReferences code.  Then, all these references should be found, and rename will update them autoamtically without having to think about it.

Comment (Cosifne) [reply src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs:140]: FindReference code already cascades these. See:
https://github.com/dotnet/roslyn/blob/6a1c377d0562781e56daf66ee4fa9d2e0c9702a0/src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs#L70
https://github.com/dotnet/roslyn/blob/6a1c377d0562781e56daf66ee4fa9d2e0c9702a0/src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ParameterSymbolReferenceFinder.cs#L68

And the rename engine gets the correct symbol and locations [here](https://github.com/dotnet/roslyn/blob/6a1c377d0562781e56daf66ee4fa9d2e0c9702a0/src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.cs#L131).

But they are filtered out by this function.

Comment (Cosifne) [reply src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs:140]: e.g. `ShouldIncludeSymbolAsync` in this file.

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs:140]: oh.  niteresting.  thanks for the clarification.

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb [hunk 2]
```diff
@@ -2383,5 +2383,48 @@ class [|C|]
...
         End Function
 
+        <WpfTheory>
+        <CombinatorialData, Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Sub RenameRecordParameter(host As RenameTestHost)
+            Using result = RenameEngineResult.Create(_outputHelper,
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true">
+                            <Document>
+                                public record MyRecord(string [|$$MyProperty|]);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/Rename/InlineRenameTests.vb:2393]: can you have a test whre this is `MyRecord<T>`, this is `T MyProperty` and you're renaming at an instantiation point where yo uhave a `MyRecord<int>`?  If you do that and it works (which i give you 50/50 odds on) then i can sign off.  tnx.

File: src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs [hunk 3]
```diff
@@ -121,6 +128,18 @@ static bool IsConstructorForType(ISymbol possibleConstructor, ISymbol possibleTy
             }
         }
 
+        private static bool IsPropertyGeneratedFromPrimaryConstructorParameter(
+            ISymbol propertySymbol, ISymbol parameterSymbol, CancellationToken cancellationToken)
+        {
+            if (parameterSymbol is IParameterSymbol parameter)
+            {
+                var generatedPropertySymbol = parameter.GetAssociatedSynthesizedRecordProperty(cancellationToken);
+                return propertySymbol.Equals(generatedPropertySymbol);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs:137]: not sure if we need to sprinkle in .OriginalDefinition here in generic cases.  see test recommendation above.

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb (hunk 2 above)
Comment (Cosifne) [reply src/EditorFeatures/Test2/Rename/InlineRenameTests.vb:2393]: Added.

File: src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs:140]: nit: can be written as:

`=> parameterSymbol is IParameterSymbol parameter && propertySymbol.Equals(parameter.GetAssociatedSynthesizedRecordProperty(cancellationToken));`

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**